### PR TITLE
[IMP] Propose to close unanswered PR after 6 months

### PR DESCRIPTION
**Pull request by max3903**
We have pull requests open for more than a year and no activity on them.

Grace period:
* 2 months : 2 votes (@moylop260, @elicoidal)
* 3 months : 2.5 votes (@nhomar, @dreispt, @clonedagain)
* 6 months : 4.5 votes (@gurneyalex, @bealdav, @max3903, @gdgellatly, @clonedagain)
* 1 year : 1 vote (@rvalyi)

### CODE REVIEW

**Comment by moylop260** [comment]
6 months is too much.
IMHO should be 2 months.
What about of continue the work by someone else?

**Comment by max3903** [comment]
@moylop260 so let's rephrase it to "* there is no activity for 2 months"

**Comment by rvalyi** [comment]
Hello, on my side I would let something like 1 year. Why? Because we are all small companies working for small customers and hence many things can happen and let the work unfinished. But making it visible there is something that is unfinished is IMHO much better than hidding it. We often investigate ideas and modules from past versions, these ideas are not obsolete in just 3 or even 6 months. Well just my opinion, curious why you want something so short.

**Comment by gdgellatly** [comment]
I think it needs splitting (which I see the code has done, just not the description on this PR).

If the creator of the Pull Request has not responded within 30 days to requests, close it / label it.  If people have taken time to review their work, then they should respond promptly.  Maybe with a reminder after 21 days (or basically at least a weeks notice before).  Or at least a tag saying, awaiting contributor update so as not to waste reviewers time.

If no one has reviewed a contribution for 30 days, then I think the maintainers of that repo should escalate in some way.  Either by requesting a review or closing it or adding a priority label or whatever.

I would not like to see PR's closed just because no one reviewed them.

I chose 30 days because open PR's slowdown all other PR's as well.  Also often what happens is that the contribution will need reworking soon after.  product-attribute repo is a good example where we have essentially 4 open pull requests for much the same thing done slightly differently. 

The contributor can always reopen it later when they have time.  IMHO 6 months is too long, that is half a release cycle.  Lets say after 179 days a contributor comes back and makes changes.  How much does the original reviewers really remember, it is wasting their time as they basically have to start again.

**Comment by rvalyi** [comment]
May be some kind of "archive" label then? So that maintainer doesn't wate time but features attempts are still searchable and reusable even unfinished.

**Comment by elicoidal** [comment]
in favor of two months for no answer. Contributors need to be active when they propose their work to OCA.
Nevertheless we see that many PR are simply not acknowledged by the community and shouldnot be closed if not first contact has been made

**Comment by blaggacao** [comment]
As most of the time, I'm with Graeme. It's thought through:

Although I'm in favor of closing at some point, because this enhances overall focus. And PR that are incentivized to be *efficiently solved* bring extra value to the ecosystem overall.

1. Stage: Creator does not interact within 15 days after interaction of third party. -> Bot-Message: "Your PR is marked for automatic closing, due to inactivity for the last 15 days. Please promote the efficient completion of the PR cycle, by providing additional information, reviewing etc.pp. If there is no interaction for the next 30 days at all, the bot will clear the issue."
2. Stage: After 30 additional days: Label with "SLUMBERING" and close (to get the dashboard clean). Committer or maintainer can revieve at any point in time by reopen it. A third party, that want to retake on it can clone and do a new PR request, referencing the old one.

**Comment by blaggacao** [comment]
Having one module = one repo, maintainers  responsibilities could be more fine grained and would be clearer (and potentially narrower)

**Comment by dreispt** [comment]
There are two possible escalation levels: a) Label it, b) Close it.

For a), it's already in our workflow: based on authors feedback, or lack of it, we label PRs as "Work In Progress" or "Needs Fixing". This is enough for reviewers to know they can skip those reviews.

For b) closing, I personally dislike having open PRs that are not active. On the other hand I also agree that it's good to have that code there, and that it can be salvaged at any time.
If we close idle PRs it's important to keep those tags, to distinguish them from over superceded PRs.
When searching for Issues we also look at closed ones, so I don't see an issue on that when looking for PRs.

I'm for allowing maintainers to be allowed to close PRs idle for 3 months.
But it should not be automatic and a discussion on the closing action should be at least started in the PR.

**Comment by bealdav** [comment]
6 months is perfect.
In big projects, we need time to use a dev in production. Production give maturity on PR.

**Comment by gurneyalex** [comment]
:+1: for 6 months

**Comment by gdgellatly** [comment]
Well seeing all the guys who do the most work say 6 months then I guess who am I to argue.

**Comment by dreispt** [comment]
I favoured 3 months, but am OK with 6 months also. No need to rush.
But please make it clear that this is to be applied at reviewers discretion - it should never be automatic.

**Comment by gurneyalex** [comment]
:+1:

**Comment by blaggacao** [comment]
I'm not sure if http://www.codetriage.com/ could help with this. It is a webservice, that sends you one open issues and PR. I suggest to incorporate a recommendation to use this tool as well. (if need be)

**Comment by moylop260** [comment]
:+1: 
https://github.com/savoirfairelinux/maintainer-tools/commit/d765bf96b388445eaede0f88f385bbf073d34810

**Comment by ajite** [comment]
I am in favor of the 6 months rule.

**Comment by moylop260** [comment]
The winner is: `6 months`

@all 
Can you review the diff of this PR?

Note: IMHO `automate` is offtopic here.

**Comment by max3903** [comment]
Looking for someone to merge here. Thanks.

**Comment by osvalr** [comment]
I support 2 months, but what about before closing it, put a label on it as "Lack of Activity" or something referring that has no activity in the last X period of time, I understand what @dreispt said about "Work In Progress" or "Needs Fixing", but IMHO none of them tell me about a certain PR has no activity at all.

**Comment by dreispt** [comment]
@osvalr GitHub keeps a last updated date, you can refer to it.
